modules/core/src/matmul.simd.hpp

```diff
@@ -2476,6 +2476,45 @@ double dotProd_8s(const schar* src1, const schar* src2, int len)
             i += blockSize;
         }
     }
+#elif CV_MSA
```

Comment:
`#if CV_SIMD` blocks this code execution here too (and compilation). The same note is about `#elif CV_NEON` code block (dead code too)

Reply:
@mipsopen-fwu Have you compared performance of this MSA specific implementation against generic UI implementation?

Reply:
@terfendail 
This CV_MSA related code block was added by mistake when I merged msa implementations from 4.0.1 branch to 4.1.1 and 3.4 branch.  On our pervious 4.0.1 development branch, this MSA specific implementation doesn't exist. It's correct to remove from the pull request.

Even though, in order to make clear the performance difference, I did MatType_Length_dot test  by setting testparam=(8UC1, 1024) and samples=15.  From the result, we can see that the performance of generic UI implementation(using CV_SIMD) is better against CV_MSA specific implementation which was added by mistake in this code block.
The following is the test result:
1) The  total running time for CV_MSA implementation is:
[----------] 18 tests from MatType_Length_dot (2177 ms total)
2) The total running time for CV_SIMD implementations is:
[----------] 18 tests from MatType_Length_dot (618 ms total)